### ISSUE TITLE
microsoft excel is waiting for another application to complete an ole action

### ISSUE THREAD

**Issue by Anubis93** [xlwings/xlwings#1451]
I wrote a udf function, which does a time series prediction. To do so I initialize multiple models (TES, ARIMA, etc.) then apply a grid search approach to do hyperparameter tuning and than select the best model for the prediction. 

If I run the function without hyperparamter tuning I have no problem, the moment I tell my function to apply hyperparameter tuning I get the followin error: 

_microsoft excel is waiting for another application to complete an ole action_

I guess the reason for this is the time the function needs to run completly. I read through some GIT issues and found out that app.display_alerts = False should help to solve the problem. So I tried to add this in my UDF without access

@xw.func
@xw.arg('data', pd.DataFrame, header = True, index = False, expand = 'table')
@xw.arg('forecast_horizon', numbers = int)
@xw.arg('period_len', numbers = int)
@xw.ret(expand = 'table')
def gravity_forecast(data, label_name, index_name, freq, forecast_horizon, period_len = 12, hyper_tuning = True, fitted = True, return_kpi = True, eval_kpi = "mae"): 
    '''Beschreibung Funktion'''
    xw.Book.caller().app.display_alerts = False
    ## Read in and prepare the data
    df = data.copy()
    df.index = pd.to_datetime(df[index_name], format = '%Y-%m-%d')
    df = df.drop([index_name], axis = 1)
.....

Does maybe someone has an idea what I did wrong or how to solve my problem ? 

Using: Windows, Python 3.8.3, xlwings 0.20.8

**Comment by ericbmoreira** [comment]
Can you use the asynchronous method? https://docs.xlwings.org/en/stable/udfs.html#asynchronous-udfs

I've run into the OLE issues but have not dug into them because they are infrequent.

**Comment by Anubis93** [comment]
@ericbmoreira I also thought that this maybe will do the job, but unfortunately it doesn't help.

**Comment by fzumstein** [comment]
I am not quite sure why `display_alerts` wouldn't work. For testing purposes, you could go into the VBA wrapper code (in the `xwlings_udfs` module and put in a `Application.DisplayAlerts = False` at the beginning of the function and see if that resolves it (this isn't meant as a permanent fix though, just for testing).

**Comment by Anubis93** [comment]
Sorry if I am a little slow on the uptake, but you mean that I should replace _xw.Book.caller().app.display_alerts = False_ with 
_xw.Book.caller().Application.DisplayAlerts = False_ right?

**Comment by fzumstein** [comment]
No, open the VBA editor with `Alt+F11`  and there, in the autogenerated VBA code in the VBA module `xlwings_udfs` add this line.

**Comment by Anubis93** [comment]
Thank you for this fast response, I added this line and I still get the OLE message.

**Comment by fzumstein** [comment]
Ok, you could try the last comment in this thread: https://github.com/xlwings/xlwings/issues/120 i.e. paste it in a VBA module, and call `SwitchOffOLEMessage` before running your UDF.

**Comment by Anubis93** [comment]
Thank you, with this functions is was able to run my code!

**Comment by fzumstein** [comment]
Great, thanks for letting me know! I'll work on a fix for this then.

**Comment by Anubis93** [comment]
Just one last comment, in case someone goes through this issue and does not really understand how to make use of the functions in #120: 

1. As fzumstein said go to the VBA Window (Win: Alt+f11)
2. Add a new module
3. Go to modul > xlwings_udf and add to the VBA wrapper of the function which causes the OLE Message the following:
Function [Function Name]([attributes])
SwitchOffOLEMessage
<some other code lines>
SwitchOnOLEMessage
End Function


@fzumstein thank you again

**Comment by fzumstein** [comment]
while I can disable the messages, I didn't manage to reenable them again except for a restarting Excel

**Comment by Thanasisleo** [comment]
Hi,

I think I have stuck on the same error. 

I tried to the `Application.DisplayAlerts = False`, but I still get an error:

```
  File "n:\*\utils.py", line 313, in checkWb
    activeWb = xw.books.active
  File "D:\Apps\Continuum\anaconda3\envs\py37\lib\site-packages\xlwings\main.py", line 2744, in active
    return Book(impl=self.impl.active)
  File "D:\Apps\Continuum\anaconda3\envs\py37\lib\site-packages\xlwings\main.py", line 2861, in impl
    return apps.active.books.impl
  File "D:\Apps\Continuum\anaconda3\envs\py37\lib\site-packages\xlwings\main.py", line 366, in books
    return Books(impl=self.impl.books)
  File "D:\Apps\Continuum\anaconda3\envs\py37\lib\site-packages\xlwings\_xlwindows.py", line 385, in books
    return Books(xl=self.xl.Workbooks)
  File "D:\Apps\Continuum\anaconda3\envs\py37\lib\site-packages\xlwings\_xlwindows.py", line 310, in xl
    self._xl = get_xl_app_from_hwnd(self._hwnd)
  File "D:\Apps\Continuum\anaconda3\envs\py37\lib\site-packages\xlwings\_xlwindows.py", line 220, in get_xl_app_from_hwnd
    ptr = accessible_object_from_window(child_hwnd)
  File "D:\Apps\Continuum\anaconda3\envs\py37\lib\site-packages\xlwings\_xlwindows.py", line 190, in accessible_object_from_window
    byref(IDispatch._iid_), byref(ptr))
  File "_ctypes/callproc.c", line 933, in GetResult
OSError: [WinError -2147467259] Unspecified error
```

I am running xlwings: 0.15.8 and these are the configuration settings: 

Interpreter | python
PYTHONPATH | N:\*\Source
UDF Modules |  
Debug UDFs | FALSE
Log File | N:\*\Log.txt
Use UDF Server | TRUE
Conda Base | D:\Apps\Continuum\anaconda3
Conda Env | py37

**Comment by fzumstein** [comment]
Are you using RunPython or a UDF?

**Comment by Thanasisleo** [comment]
> Are you using RunPython or a UDF?

RunPython

**Comment by fzumstein** [comment]
In that case I would just recommend you to switch `Use UDF Server` to `FALSE` as the overhead doesn't make a huge difference given that your call seems to take longer than 90seconds anyway.

**Comment by Thanasisleo** [comment]
Thing is that I like the feature to have the terminal open in order to see at which stage the code is. But if setting it to `FALSE `will fix the issue, then I will do it. 

Also I had a short script and I wanted it to run fast, that is why I used the `Use UDF Server` to `TRUE `but now I see that even with false it runs quite fast too.

**Comment by fzumstein** [comment]
You can still have the terminal open via the `Show Console` setting (new in 0.19.5), it just closes at the end of the program with `Use UDF server = False`

**Comment by Thanasisleo** [comment]
> You can still have the terminal open via the `Show Console` setting (new in 0.19.5), it just closes at the end of the program with `Use UDF server = False`

OK, I will try it.
